compiler/src/main/scala/org/scalajs/core/compiler/PrepWasPublicBeforeTyper.scala

```diff
@@ -0,0 +1,72 @@
+/* Scala.js compiler
+ * Copyright 2013 LAMP/EPFL
+ * @author Nicolas Stucki
+ */
+
+package org.scalajs.core.compiler
+
+import scala.tools.nsc
+import nsc._
+
+/**
+ * @author Nicolas Stucki
+ */
+abstract class PrepWasPublicBeforeTyper extends plugins.PluginComponent with transform.Transform
```

Comment:
That doc should also explain that we do this to work around SI-something from scalac.